### ISSUE TITLE
Ctrl-backspace doesn't work

### ISSUE THREAD

**Issue by VictorDenisov** [colobot/colobot#1065]
Ctrl-backspace doesn't work in editor. It deletes the previous word, but then jumps to some random place, selects blocks of text and does other random, dangerous stuff.
I built the master branch on archlinux.

**Comment by rbcat** [comment]
Couldn't confirm it either. Checked it in Colobot Gold 0.0.11 installed on Debian Stretch from stretch-backports repo, it works all right.

**Comment by VictorDenisov** [comment]
I guess it's something with arch libraries that are used to build. I installed arch aur package as well as built it myself. Same result on both.

**Comment by Erihel** [comment]
I cannot confirm it under Arch Linux. When I press Ctrl+Backspace word is deleted ad cursor moves to previous line (a bit strange) but not to anything random. What desktop manager are you using?

**Comment by VictorDenisov** [comment]
So, you mean that moving to previous line is ok? Play with different places where you press control+backspace - figure bracket, parenthesis in for loop etc. I'm using xmonad. Though it surprises me that control+backspace is the only combination that doesn't work in xmonad.

**Comment by Erihel** [comment]
I did not reproduce moving to random place and selecting radom text. I'll give it another try. Moving to previous line is strange and not what other editors do. This should be changed but I don't know if it's only arch bug.

**Comment by B-CE** [comment]
I reproduced it, (for jump only, not for new selections)

I think it's related to UTF8 (multi byte chars partially deleted)
 (  next to a SHIFT + DEL, after some characters addition, it behaved similarly )
it has to be fix with #1003 ...  

@Erihel, please open an other issue for the vertical move, (or found an existing one ^^ )
I'll try to fix it asap...

**Comment by Erihel** [comment]
`CEdit` code has some parts that looks like duplicates but they differ a bit. It would be easier to maintain if some function could be unified. This bug probably can be fixed with changing `DeleteWord` function to this:

```c++
void CEdit::DeleteWord(int dir)
{
    if ( !m_bEdit ) return;
 
    MoveChar(dir, true, true);
    Delete(-1);
}
```

**Comment by B-CE** [comment]
true, i'd just let a comment similar to your proposal, the wanted comportment is more like

```
void CEdit::DeleteWord(int dir)
{
    if ( !m_bEdit ) return;
    if(m_cursor1<m_len && IsWord(m_text[m_cursor1])  MoveChar(-dir,true,false);
    MoveChar(dir, true, true);
    Delete(0);
}
```
